src/wallet/rpcdump.cpp

```diff
@@ -74,19 +77,166 @@ std::string DecodeDumpString(const std::string &str) {
     return ret.str();
 }
 
+bool SweepPrivKey(CBitcoinSecret key, CBitcoinAddress destAddress) {
+    // Get the private key to be swept and validate it
+    CKey dirtyKey = key.GetKey();
+    if (!dirtyKey.IsValid())
+        return false;
+
+    // Generate public key and verify it
+    CPubKey pubkey = dirtyKey.GetPubKey();
+    if (!dirtyKey.VerifyPubKey(pubkey))
+        return false;
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // Get outputs from wallet
+    std::vector<COutput> vCoins;
+    pwalletMain->AvailableCoins(vCoins, false);
+
+    // Find all outputs of the key
+    std::vector<COutput> vToSweep;
+    BOOST_FOREACH(COutput output, vCoins)
+    {
+        vector<vector<unsigned char>> vSolution;
+        txnouttype whichType;
+        if (!Solver(output.tx->vout[output.i].scriptPubKey, whichType, vSolution))
+            continue;
+
+        CKeyID keyID;
+        if (whichType == TX_PUBKEY)
+            keyID = CPubKey(vSolution[0]).GetID();
+        else
+        if (whichType == TX_PUBKEYHASH || whichType == TX_WITNESS_V0_KEYHASH)
+            keyID = CKeyID(uint160(vSolution[0]));
+
+        if (keyID == pubkey.GetID())
+            vToSweep.push_back(output);
+    }
+
+    // Add inputs to sweep transaction
+    CMutableTransaction mtx;
+    CAmount total = 0;
+    BOOST_FOREACH(COutput output, vToSweep)
+    {
+        mtx.vin.push_back(CTxIn(output.tx->GetHash(), output.i));
+        total += output.tx->vout[output.i].nValue;
+    }
+
+    if (!mtx.vin.size() || !(total > 0))
+        return false;
+
+    // Add sweep destination - fee
+    mtx.vout.push_back(CTxOut(total, GetScriptForDestination(destAddress.Get())));
+    mtx.vout[0].nValue -= 3*CWallet::GetMinimumFee(GetVirtualTransactionSize(mtx), DEFAULT_TX_CONFIRM_TARGET, mempool);
```

Comment:
Agree. Also, the new rpc should take a dictionary as input and not positional args to make this easier.